### ISSUE TITLE
3.9.2 latest update crashed site

### ISSUE THREAD

**Issue by PascalBG** [woocommerce/woocommerce#25669]
Did the update, everything blank. Frontend, back-end. Everything. Blank.

**Comment by PascalBG** [comment]
I can also not login to wordpress.org... If I replace a 1 by 1 the files with previous plugin version, would that be ok to try to recover? No idea what that has to do with not being able to login to .org... (site cannot be reached it says. )

**Comment by PascalBG** [comment]
It is following code in theme functions.php that is the cause:
`//	if( is_product() && $translated == 'Beoordelingen' && $domain == 'woocommerce' ) {
//		$translated = 'Laat als eerste een ervaring/review achter!';
//	`}``

**Comment by peterfabian** [comment]
Hi @PascalBG .

Sorry that you're experiencing problems. Based on what you're saying, it seems like the theme might be performing some actions that are causing these problems. Would you mind sharing the [system status report](https://docs.woocommerce.com/document/how-to-review-your-woocommerce-system-status/) if you manage to get your site up and running for us to better understand what theme and plugins you are using? 

Can you share the relevant sections from your PHP error log by any chance, if you know where to find it?

Thanks!

**Comment by PascalBG** [comment]
The system status report link is to docs.woocommerce.com, link does not work with me just like login.wordpress.org does not work. Cannot do what you ask since cannot reach source to see what it is.

The code is added in my previous post. It is a customisation for the text that shows on the tab reviews when there is no review.

Something in the new update broke it.

500 errors
php log pointed exactly to 	if( is_product()

By commenting out the specific section with 	if( is_product() everything works again.

Very easy to reproduce normally.
IT is fixed now. I have explained here what it is.

**Comment by tammullen** [comment]
Hi @PascalBG,

Thanks for getting back to us and glad to hear it sounds like you managed to fix your issue.

Unfortunately without more details of the issue itself we are unable to confirm if there is a bug. 

If you can try again to access the [system status report using these instructions](https://docs.woocommerce.com/document/how-to-review-your-woocommerce-system-status/)
and share it with us.

If you are having trouble connecting to docs.woocommerce.com or wordpress.org this might be an issue with your internet connection. I suggest trying again. You can also validate if they are available by using a 3rd party site that can check site availability such as https://downforeveryoneorjustme.com/docs.woocommerce.com

To help determine if you have found a bug, please follow the steps listed in **Isolating the problem** section of our [bug report template](https://github.com/woocommerce/woocommerce/issues/new?assignees=&labels=&template=Bug_report.md&title=). If you confirm the bug on your end, please provide us with clear steps to reproduce the issue as well as the system status report.

**Comment by tammullen** [comment]
Hi @PascalBG,

As a part of this repository’s maintenance, I am closing this issue due to inactivity. Please feel free to comment on it in case we missed something. We’d be happy to take another look.

**Comment by Miqi180** [comment]
I can confirm this definitely **is** an issue with WooCommerce 3.9.2! If conditional tags (in my case **is_woocommerce()**) are present in the beginning of the functions.php file of the theme, the update crashes the site with an http 500 error. Fortunately, PascalBG's fix also worked for me, i.e. 1) outcomment the code containing the conditional tag 2) install 3.9.2 3) reactivate the outcommented code. This problem seems to only arise during the update of WC.

**Comment by peterfabian** [comment]
Thanks, I now see what you're saying: some themes are calling `is_woocommerce()` or `is_product()` functions in the theme's functions.php file which causes a fatal error during the upgrade. However, I wasn't able to reproduce this by following these steps:
- install WC 3.9.2
- install Storefront
- add a call to `is_woocommerce()` in `wp-content/themes/storefront/functions.php`
- upgrade to WC 3.9.3

Would you mind sharing your system status or just the theme name so that we can try to reproduce this please?

**Comment by PascalBG** [comment]
I use a child theme of storefront.
in the functions.php the code I earlier mentioned is around line 150, after several other lines of code and functions.
All works now.

**Comment by Miqi180** [comment]
Hi.

So I just updated to WC v. 3.9.3 on our staging site which uses the same theme as the production site without any problems, so the update bug from v. 3.9.2 seems to have been fixed. 

Theme: 
`/*
Theme Name: EBookStore
Theme URI: http://books.kodeforest.net/library/
Description: The E Book Store Wordpress Theme
Author: KodeForest
Author URI: http://www.kodeforest.com
Version: 2.7
License: 
License URI: http://www.gnu.org/licenses/gpl-2.0.html
Tags: featured-images, theme-options
Text Domain: e-book-store
*/`

The code in the functions.php file which caused the crash:

`add_filter( 'gettext', 'theme_sort_change', 20, 3 );
    function theme_sort_change( $translated_text, $text, $domain ) {

        if ( **is_woocommerce()** ) { //this line of code was the problem

            switch ( $translated_text ) {

            ...

        }

        return $translated_text;
    }`

What I find very strange, though is that many other procedures in our functions.php file contain WC conditional tags, Here's the complete list:

`
is_woocommerce() // which is the first occurence a WC conditional tag and which caused the crash
is_shop()
is_product_category()
is_product()
`

Why would it suffice to outcomment only the first occurence of a WC conditional tag in the functions.php file when several are present, especially considering `is_product()` crashed PascalBG's site!?

**Comment by peterfabian** [comment]
Thanks for the updates. Seems strange, as nothing in 3.9.3 should change the behaviour with respect to this problem, as far as I know. 

I'll close this one for now, then, since you report not having the problem anymore. Please feel free to comment/open a new issue if the problem comes back.

**Comment by PascalBG** [comment]
Hi, the problem occurred when updating from 3.9.1 to 3.9.2
I just updated from 3.9.2 to 3.9.3 without any problems

**Comment by PascalBG** [comment]
Again same problem! Updated to 4.0.0
Exactly same problem.
Deleted functions.php in child theme and could reach the site again.
In the admin I found out woocommerce indeed was updated BUT not activated. So I activated it, everything works fine after db update
Uploaded functions.php again and everything working.

**Comment by peterfabian** [comment]
Hi, 

I'm not exactly sure how this happens on your site, but at some point, the code you added runs before WooCommerce is active. The `gettext` filter is a very broad filter that runs for every string on your site, thus it can happen that in some special circumstances, it runs before WC is active (even though, in general, themes should be loaded later than the plugins). 

If you want to avoid this, you can try to swap the conditions in your custom code, e.g. 

instead of 
```

if( is_product() && $translated == 'Reviews' && $domain == 'woocommerce' ) {
		$translated = 'Ooops... No reviews yet. Please leave one!';
	}

```

use sth like this:

```

if( $domain == 'woocommerce' && $translated == 'Reviews' && is_product() ) {
		$translated = 'Ooops... No reviews yet. Please leave one!';
	}

```

That should ensure that `is_product()` is only run if the $domain is `woocommerce`, which should be true only after the plugin has loaded.
